ENGAGEHF/Education/VideoManager.swift

```diff
@@ -34,24 +35,27 @@ final class VideoManager: Module, EnvironmentAccessible, DefaultInitializable {
             return
         }
 #endif
-        
-        authStateDidChangeListenerHandle = Auth.auth().addStateDidChangeListener { [weak self] _, user in
-            if let user {
-                Task {
-                    guard let collections = await self?.getVideoSections(user: user) else {
+
+        if let accountNotifications {
+            notificationTask = Task.detached { @MainActor [weak self] in
+                for await event in accountNotifications.events {
+                    guard let self else {
```

Comment:
Should we also check for Task cancellation in here? Applies to the other usages of this as well.

Reply:
I think the AsyncStream automatically checks for cancellation and just completes, if I’m not wrong. But I’ll re-review that 👍